qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1520A.py

```diff
@@ -161,6 +464,206 @@ def abort(self):
+                depend on the output signal frequency.
+
+            hold_delay: Hold time (in seconds) that is the wait time after
+                starting measurement and before starting delay time for the
+                first step 0 to 655.35, with 10 ms resolution.
+
+            delay: Delay time (in seconds) that is the wait time after
+                starting to force a step output and before starting a step
+                measurement.
+
+            step_delay: Step delay time (in seconds) that is the wait time
+                after starting a step measurement and before starting to
+                force the next step output. 0 to 1, with 0.1 ms resolution.
+                If  step_delay is shorter than the measurement time,
+                the B1500 waits until the measurement completes, then forces
+                the next step output.
+
+            trigger_delay: Step source trigger delay time (in seconds) that
+                is the wait time after completing a step output  setup and
+                before sending a step output setup  completion trigger. 0 to
+                delay, with 0.1 ms resolution.
+
+            measure_delay: Step measurement trigger delay time (in
+                seconds) that is the wait time after receiving a start step
+                measurement trigger and before starting a step measurement.
+                0 to 65.535, with 0.1 ms resolution.
+
+            abort_enabled: Boolean, enables or disables the automatic abort
+                function for the CV sweep measurement.
+
+            sweep_mode: Linear sweep, log sweep, linear 2 way sweep or
+                log 2 way sweep
+
+            volt_monitor: Boolean
+
+        """
+
+        self.root_instrument.enable_channels(self.channels)
+        self.adc_mode(adc_mode)
+        self.adc_coef(adc_coef)
+        self.frequency(freq)
+        self.voltage_ac(ac_rms)
+        self.cv_sweep.sweep_auto_abort(abort_enabled)
+        self.cv_sweep.post_sweep_voltage_cond(post_sweep_voltage_cond)
+        self.cv_sweep.hold(hold_delay)
+        self.cv_sweep.delay(delay)
+        self.cv_sweep.step_delay(step_delay)
+        self.cv_sweep.trigger_delay(trigger_delay)
+        self.cv_sweep.measure_delay(measure_delay)
+        self.sweep_mode(sweep_mode)
+        self.sweep_start(v_start)
+        self.sweep_end(v_end)
+        self._sweep_steps = n_steps
+        self.sweep_steps(self._sweep_steps)
+        self.measurement_mode(constants.MM.Mode.CV_DC_SWEEP)
+        self.impedance_model(imp_model)
+        self.ac_dc_volt_monitor(volt_monitor)
+        self.ranging_mode(ranging_mode)
+        self.measurement_range_for_non_auto(fixed_range_val)
+
+        err = self.root_instrument.error_message()
+        if err == '+0,"No Error."':
+            self.setup_fnc_already_run = True
+        return err
+
+    @staticmethod
+    def parse_sweep_data(raw_data: str) -> tuple:
```

Comment:
Beautify this ; use re

Reply:
One more issue: if ac_dc_volt_monitor is set to True. Current rudimentary parsing will fail. The the output is: 
```
'NCC+6.01120E-13,NCD+2.03051E-02,NCV+29.9666E-03,NCV-1.00013E+00,NCC+5.82716E-13,NCD-5.69785E-02,NCV+29.9663E-03,NCV-0.50018E+00,NCC+5.78574E-13,NCD-1.28436E-02,NCV+29.9667E-03,NCV-0.00044E+00,NCC+5.45751E-13,NCD+1.19937E-02,NCV+29.9663E-03,NCV+0.49950E+00,NCC+5.92740E-13,NCD+2.01654E-02,NCV+29.9671E-03,NCV+0.99937E+00'
```

Reply:
if ac_dc_volt_monitor is set to False; output is :
```
NCC+5.89863E-13,NCD-1.81730E-02,NCC+5.89799E-13,NCD+1.34570E-02,NCC+5.62257E-13,NCD-2.60953E-02,NCC+5.71111E-13,NCD+3.65428E-03,NCC+5.68360E-13,NCD+1.28519E-02'
```

Reply:
decided:
* to write a separate version of parse_fmt_1_0_* that support only CMU data (this is an quick-easy to do now as opposed to implementing parsers for all the cases of `FMT 1,0`)
* to use ``ac_dc_volt_monitor`` value to decide between two parsers (one for 2 data points per block, another one for 4)
* we do want to take status values, and make that available to the user (see `compliance` method from sampling measurement) (perhaps, make an enum for the statuses ;) )

Reply:
- the separate parser was not needed parse_fmt_1_0_*. I just used the old and added extra logic in get_raw.
"we do want to take status values, and make that available to the user (see compliance method from sampling measurement) (perhaps, make an enum for the statuses ;) )" Will clarify this point offline